common/src/main/java/org/mvndaemon/mvnd/common/Environment.java

```diff
@@ -163,6 +163,12 @@ public enum Environment {
      * non-native clients and is useful mostly for debugging.
      */
     MVND_NO_DAEMON("mvnd.noDaemon", "MVND_NO_DAEMON", Boolean.FALSE, OptionType.BOOLEAN, Flags.DISCRIMINATING),
+
+    /**
+     * If <code>true</code>, the daemon will not use its in-memory metadata cache and instead re-read the
+     * metadata from the pom.xml files in the local repository. This is mostly useful for testing purposes.
+     */
+    MVND_NO_MODEL_CACHE("mvnd.noModelCache", null, Boolean.FALSE, OptionType.BOOLEAN, Flags.NONE),
```

Comment:
I wonder whether we should use `Flags.DISCRIMINATING`? Is the `createCache()` method called once for a given daemon instance or once per Maven build? 

If it's once for daemon, I'd say it should be `Flags.DISCRIMINATING`. Otherwise it could happen that the user starts the daemon with `mvnd -Dmvnd.noModelCache` and all subsequent `mvnd` invocations with *or without* `-Dmvnd.noModelCache` will be served by the same daemon instance that has model cache off. That would not be correct, would it?

`Flags.DISCRIMINATING` will make all calls with `-Dmvnd.noModelCache` be served by a cacheless instance while all calls without `-Dmvnd.noModelCache` will be served by some other daemon instance that has the cache enabled.

Reply:
The `createCache` method is called every build, so it doesn't have to be discriminating. It doesn't hurt either, I don't mind :)

Reply:
If it is called every build, then no need to change anything. Thanks!